src/google/protobuf/compiler/js/js_generator.cc

```diff
@@ -2406,17 +2406,32 @@ void Generator::GenerateClassFieldToObject(const GeneratorOptions& options,
     }
     if (field->type() == FieldDescriptor::TYPE_ENUM) {
       // For enums, emit the key name (not the ordinal).
-      printer->Print(
-        "Object.keys($enumprefix$$enumname$).find((key) => $enumprefix$$enumname$[key] === ",
-        "getter", JSGetterName(options, field),
-        "enumprefix", GetEnumPathPrefix(options, field->enum_type()),
-        "enumname", field->enum_type()->name());
+      if (!field->is_repeated()) {
+        printer->Print(
+          "Object.keys($enumprefix$$enumname$).find((key) => $enumprefix$$enumname$[key] === ",
+          "getter", JSGetterName(options, field),
+          "enumprefix", GetEnumPathPrefix(options, field->enum_type()),
+          "enumname", field->enum_type()->name());
+      } else {
+        printer->Print(
+          "Object.keys($enumprefix$$enumname$).filter((key) => (",
+          "getter", JSGetterName(options, field),
+          "enumprefix", GetEnumPathPrefix(options, field->enum_type()),
+           "enumname", field->enum_type()->name());
+      }
     }
 
     GenerateFieldValueExpression(printer, "msg", field, use_default);
 
     if (field->type() == FieldDescriptor::TYPE_ENUM) {
-      printer->Print(")");
+      if (!field->is_repeated()) {
+        printer->Print(")");
+      } else {
+        printer->Print(
+          " || []).includes($enumprefix$$enumname$[key]))",
+          "enumprefix", GetEnumPathPrefix(options, field->enum_type()),
+          "enumname", field->enum_type()->name());
+      }
```

Comment:
This generates 
![image](https://user-images.githubusercontent.com/7275468/129628036-f6aa612f-223b-48c5-b990-54d7dd0c92df.png)